pkg/database/migrate.go

```diff
@@ -306,6 +309,40 @@ func checkAndMigrate(migrate bool, dbProvider DBProvider) error {
 				}); err != nil {
 					return err
 				}
+				fallthrough
+
+			case "5d042539be4f":
+				log.Info("Migrating database to FastTrackML schema e0d125c68d9a")
+				// We need to run this migration without foreign key constraints to avoid
+				// the cascading delete to kick in and delete all the runs.
+				if err := runWithoutForeignKeyIfNeeded(func() error {
+					if err := db.Transaction(func(tx *gorm.DB) error {
+						if err := tx.AutoMigrate(&Namespace{}); err != nil {
+							return err
+						}
+						if err := tx.Migrator().AddColumn(&Experiment{}, "NamespaceID"); err != nil {
+							return err
+						}
+						if err := tx.Migrator().CreateConstraint(&Namespace{}, "Experiments"); err != nil {
+							return err
+						}
+						if err := tx.Migrator().AlterColumn(&Experiment{}, "Name"); err != nil {
+							return err
+						}
+						if err := tx.Migrator().CreateIndex(&Experiment{}, "idx_namespace_name"); err != nil {
+							return err
+						}
```

Comment:
We're still missing the migration for Apps

Reply:
you are right. done.